### PR TITLE
Retune radix sort, run length encoding, reduce by key, scan, select if, and histogram for SM70 and SM80

### PR DESCRIPTION
**Pull request by brycelelbach**

### CODE REVIEW

File: cub/device/dispatch/dispatch_radix_sort.cuh [hunk 1]
```diff
@@ -796,8 +796,8 @@ struct DeviceRadixSortPolicy
         typedef AgentRadixSortDownsweepPolicy <256, 19, DominantT,  BLOCK_LOAD_DIRECT, LOAD_LDG, RADIX_RANK_MEMOIZE, BLOCK_SCAN_WARP_SCANS, SINGLE_TILE_RADIX_BITS>          SingleTilePolicy;
 
         // Segmented policies
-        typedef AgentRadixSortDownsweepPolicy <192, 39, DominantT,  BLOCK_LOAD_TRANSPOSE, LOAD_DEFAULT, RADIX_RANK_MEMOIZE, BLOCK_SCAN_WARP_SCANS, SEGMENTED_RADIX_BITS>     SegmentedPolicy;
-        typedef AgentRadixSortDownsweepPolicy <384, 11, DominantT,  BLOCK_LOAD_TRANSPOSE, LOAD_DEFAULT, RADIX_RANK_MEMOIZE, BLOCK_SCAN_WARP_SCANS, SEGMENTED_RADIX_BITS - 1> AltSegmentedPolicy;
+        typedef AgentRadixSortDownsweepPolicy <128, 39, DominantT,  BLOCK_LOAD_TRANSPOSE, LOAD_DEFAULT, RADIX_RANK_MEMOIZE, BLOCK_SCAN_WARP_SCANS, SEGMENTED_RADIX_BITS>     SegmentedPolicy;
+        typedef AgentRadixSortDownsweepPolicy <256, 11, DominantT,  BLOCK_LOAD_TRANSPOSE, LOAD_DEFAULT, RADIX_RANK_MEMOIZE, BLOCK_SCAN_WARP_SCANS, SEGMENTED_RADIX_BITS - 1> AltSegmentedPolicy;
     };
```
Comment (brycelelbach) [inline cub/device/dispatch/dispatch_radix_sort.cuh:801]: Maybe we should add an SM80 policy?

Comment (allisonvacanti) [reply cub/device/dispatch/dispatch_radix_sort.cuh:711]: Andy's PR adds one. Dunno if he tuned the non-onesweep parameters or just pulled them from the SM70 tunings. 

cc: @canonizer

**Comment by allisonvacanti** [comment]
DVS CL: 29273912

**Comment by allisonvacanti** [comment]
The `device_histogram.thorough` tests are failing on GV100 with the new SM70 tunings.

```
7985: CUB cub::DeviceHistogramEven (pointer) 2073600 pixels (1080 height, 1920 width, 7693-byte row stride), 8308440 1-byte h samples (entropy reduction 0), i counters, 3/4 channels, max sample 256
7985:
7985:   Channel 0: 256 bins [0, 256)
7985:
7985:   Channel 1: 128 bins [64, 192)
7985:
7985:   Channel 2: 64 bins [96, 160)
7985: CUDA error 716 [/home/av/code/src/thrust/dependencies/cub/cub/device/dispatch/../../agent/../block/../iterator/../util_device.cuh, 564]: misaligned address
7985: CUDA error 716 [/home/av/code/src/thrust/dependencies/cub/cub/device/dispatch/dispatch_histogram.cuh, 672]: misaligned address
7985: CUDA error 716 [/home/av/code/src/thrust/dependencies/cub/cub/device/dispatch/dispatch_histogram.cuh, 1048]: misaligned address
7985: Invoking DeviceHistogramInitKernel<<<1, 256, 0, 0>>>()
7985: Invoking histogram_sweep_kernel<<<{1, 320, 1}, 384, 0, 0>>>(), 5 pixels per thread, 4 SM occupancy
7985: INCORRECT: [0]: 0 != 8
7985: (/home/av/code/src/thrust/dependencies/cub/test/test_device_histogram.cu: 795)
1/1 Test #7985: cub.cpp17.test.device_histogram.thorough ...***Failed   26.55 sec
```

File: cub/device/dispatch/dispatch_radix_sort.cuh (hunk 1 above)
Comment (canonizer) [reply cub/device/dispatch/dispatch_radix_sort.cuh:711]: For the `Policy800` in my PR, I copied the parameters from `Policy700` and only tuned the onesweep-related parameters.

**Comment by allisonvacanti** [comment]
The radix sort tunings from this PR caused a perf regression, nvbug 200676467. Those parameters were merged in #204 and reverted by #237.

**Comment by senior-zero** [comment]
Most of the algorithms have been changed significantly since this PR was opened. We'll have to tune algorithms in a more systematic way. Further work is tracked by the following [issue](https://github.com/NVIDIA/cub/issues/631).

**Pull request closed without merging**